api/maven-api-xml/src/main/java/org/apache/maven/api/xml/XmlNode.java

```diff
@@ -527,7 +527,7 @@ private String toStringObject() {
                 w = addToStringField(sb, value, o -> !o.isEmpty(), "value", w);
                 w = addToStringField(sb, attributes, o -> !o.isEmpty(), "attributes", w);
                 w = addToStringField(sb, children, o -> !o.isEmpty(), "children", w);
-                w = addToStringField(sb, inputLocation, Objects::nonNull, "inputLocation", w);
+                addToStringField(sb, inputLocation, Objects::nonNull, "inputLocation", w);
```

Comment:
I agree the last variable value is not used, but I'm really skeptical about the benefit, and we're loosing ease of understanding imho.